lib/StaticNarrative/exporter/icon_util.py

```diff
@@ -0,0 +1,92 @@
+"""Icon-related functions."""
+
+import json
+from typing import Any
+
+from traitlets.config import Config
+
+from StaticNarrative import STATIC_NARRATIVE_BASE_DIR
+
+ICON_DATA_FILE = STATIC_NARRATIVE_BASE_DIR / "data" / "icons.json"
+ICON_DATA = None
+
+
+def _load_icon_data() -> None:
+    # this should access the local folder
+    global ICON_DATA
+    try:
+        with ICON_DATA_FILE.open() as icon_file:
+            ICON_DATA = json.load(icon_file)
+    except FileNotFoundError as e:
+        msg = f"File {ICON_DATA_FILE} not found."
+        raise FileNotFoundError(msg) from e
+    except json.JSONDecodeError as e:
+        msg = f"File {ICON_DATA_FILE} is not a valid JSON file."
+        raise ValueError(msg) from e
+    except Exception as e:
+        msg = f"An error occurred while loading the file {ICON_DATA_FILE}: {e}"
+        raise RuntimeError(msg) from e
+    if not ICON_DATA:
+        msg = f"No icon data found in {ICON_DATA_FILE}."
+        raise ValueError(msg)
+
+
+def get_data_icon(obj_type: str) -> dict[str, str]:
+    """Get the appropriate icon metadata for a specific object type."""
+    if not ICON_DATA:
+        _load_icon_data()
+
+    # this should not be needed, but is added in to keep python type-checking happy.
+    if not ICON_DATA:
+        msg = f"No icon data found in {ICON_DATA_FILE}."
+        raise ValueError(msg)
```

Comment:
another bit of checking - it's somewhat redundant but otherwise the type-checker throws a hissy fit.